### PR TITLE
'unable to open shell' -> direct to web help

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY
The "unable to open shell" error is returned for a number of different,
direct people to online docs (we we can update out of band of releases)
to guide them though the various solutions.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
networking

##### ANSIBLE VERSION
```
ansible 2.4.0 (unabletoopenshell 0e954dcd6e) last updated 2017/04/04 20:33:08 (GMT +100)

```

Example output
```yaml
TASK [Get facts] *********************************************************************************************************************************************************************************************************************************
fatal: [ios01]: FAILED! => {"changed": false, "failed": true, "msg": "unable to open shell. Please see: https://docs.ansible.com/ansible/network_debug_troubleshooting.html#unable-to-open-shell", "rc": 255}
	to retry, use: --limit @/home/johnb/git/ansible-inc/ansible-workspace-2/test/integration/bug.retry

PLAY RECAP ***************************************************************************************************************************************************************************************************************************************
ios01                      : ok=0    changed=0    unreachable=0    failed=1 

```

For `stable-2.3` & `devel`

### CODE REVIEW

**Comment by gundalow** [comment]
Doc's aren't currently live, though from a local build `docs/docsite/_build/html/network_debug_troubleshooting.html#unable-to-open-shell` works

**Review by Qalthos — APPROVED** [review]

File: lib/ansible/plugins/action/eos.py [hunk 1]
```diff
@@ -74,7 +74,9 @@ def run(self, tmp=None, task_vars=None):
                 display.vvvv('calling open_shell()', pc.remote_addr)
                 rc, out, err = connection.exec_command('open_shell()')
                 if not rc == 0:
-                    return {'failed': True, 'msg': 'unable to open shell'}
+                    return {'failed': True,
+                            'msg': 'unable to open shell. Please see: '
+                                   + 'https://docs.ansible.com/ansible/network_debug_troubleshooting.html#unable-to-open-shell'}
```
Comment (Qalthos) [inline lib/ansible/plugins/action/eos.py]: Missing `'rc': rc` ?

File: lib/ansible/plugins/action/eos.py [hunk 2]
```diff
@@ -74,7 +74,9 @@ def run(self, tmp=None, task_vars=None):
                 display.vvvv('calling open_shell()', pc.remote_addr)
                 rc, out, err = connection.exec_command('open_shell()')
                 if not rc == 0:
-                    return {'failed': True, 'msg': 'unable to open shell'}
```
Comment (gundalow) [inline lib/ansible/plugins/action/eos.py:77]: Not sure why on eos we don't return `rc`

**Comment by gundalow** [comment]
Shippable failures are not related

**Comment by gundalow** [comment]
Merged to `devel` and `stable-2.3` https://github.com/ansible/ansible/pull/23331

**Pull request merged**